### PR TITLE
ENG-13016 missing rows in Kafka importer when cluster is paused and resumed

### PR DESCRIPTION
**Pull request by zeemanhe**
A couple of issues contribute to the missing rows when a cluster is paused and resumed.
(1) When a cluster is paused, an importer may over commit offsets to Kafka. After importers are restarted upon the cluster's resuming, the rows which are not persisted to Volt will not be pulled over again from Kafka.
(2)It is assumed that the very first message from Kafka is alway persisted to Volt. The assumption is incorrect.
The pull request addressed these two issues and also simplified the testing benchmark.

### CODE REVIEW

File: src/frontend/org/voltdb/importclient/kafka/util/DurableTracker.java [hunk 1]
```diff
@@ -31,32 +31,47 @@
...
         this.topic = topic;
         this.partition = partition;
+        this.consumerGroup = "";
+    }
+
+    public DurableTracker(int leeway, String topic, int partition, String consumerGroup) {
+        if (leeway <= 0) {
+            throw new IllegalArgumentException("leeways is zero or negative");
+        }
+        lag = new long[leeway];
```
Comment (luyangco) [inline src/frontend/org/voltdb/importclient/kafka/util/DurableTracker.java]: For sake of the readability, `lag` can be reimplemented with a `RangeSet<Long>`, every commit insert a new range to  the RangeSet, merge with previous range if possible. Commit tracker always return the end offset of first contiguous range.

DRConsumerDrIdTracker uses the similar implementation.

File: tests/test_apps/kafkaimporter/db/procedures/InsertImportWithCount1.java [hunk 2]
```diff
@@ -38,19 +38,11 @@
 
 public class InsertImportWithCount1 extends VoltProcedure {
     public final String sqlSuffix = "(key, value) VALUES (?, ?)";
-    public final SQLStmt importInsert = new SQLStmt("INSERT INTO kafkaImportTable1 " + sqlSuffix);
-    public final SQLStmt incrementMirrorRow = new SQLStmt("UPDATE kafkamirrortable1 SET import_count=import_count+1 WHERE key = ? and value = ?");
-    public final SQLStmt insertError = new SQLStmt("INSERT into importnomatch (key, value, streamnumber) values(?, ?, 1)");
```
Comment (luyangco) [inline tests/test_apps/kafkaimporter/db/procedures/InsertImportWithCount1.java]: Why remove the insertError test?

File: src/frontend/org/voltdb/importclient/kafka10/KafkaConsumerRunner.java [hunk 3]
```diff
@@ -206,6 +209,9 @@ public void run() {
                     //The messages in next poll starts at the largest offset + 1 in the previous polled messages.
                     //Every message is polled only once.
                     ConsumerRecords<ByteBuffer, ByteBuffer> records = null;
+                    if (!m_lifecycle.shouldRun()) {
```
[the remark this answers is not in the document]
Comment (zeemanhe) [reply src/frontend/org/voltdb/importclient/kafka10/KafkaConsumerRunner.java]: This is to reduce chance of doing another poll.

File: src/frontend/org/voltdb/importclient/kafka/util/DurableTracker.java (hunk 1 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/importclient/kafka/util/DurableTracker.java]: great idea. may be done in a separate enhancement.

File: tests/test_apps/kafkaimporter/db/procedures/InsertImportWithCount1.java (hunk 2 above)
Comment (zeemanhe) [reply tests/test_apps/kafkaimporter/db/procedures/InsertImportWithCount1.java]: Removed since it is not used in the benchmark test.

**Pull request merged**